docs/plugins/call-existing-api.md

```diff
@@ -1,6 +1,168 @@
+  megabyte blob with large amounts of redundant information, it would hurt the
+  end user experience. Particularly on mobile. The same goes for cases where you
+  want to show many individual pieces of information: if a common use case is to
+  show large tables where one API request per cell is necessary, the browser
+  will quickly become swamped and you may want to consider performing
+  aggregation elsewhere instead.
+- The API can maintain interactive request/response times at your required peak
+  request rates. The end user experience will be degraded if they spend a lot of
+  time waiting for the data to arrive.
+- The API endpoint is highly available. The browser does not have builtin
+  facilities for load balancing, service discovery, retries, health checks,
+  circuit breaking and similar. If the endpoint is occasionally down even for
+  short periods of time (e.g. during deploys), end users will quickly notice.
+- The API is exposed over HTTPS (not just HTTP), and properly handles
+  [CORS](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS). These are
+  requirements that the user's browser will impose for security reasons, and the
+  requests will be rejected otherwise.
+- The API endpoint is easily reachable, in terms of network conditions, by end
+  users. This may be particularly relevant if your end users are outside of your
+  perimeter.
+- The requests do not require secrets to be passed. This limitation does not
+  apply to OAuth tokens, which the frontend can negotiate and make proper use
+  of.
+
+## Using The Backstage Proxy
+
+Backstage has an optional proxy plugin for the backend, that can be used to
+easily add proxy routes to downstream APIs.
+
+Example:
+
+```yaml
+# In app-config.yaml
+proxy:
+  '/frobs':
+    target: 'http://frobs.example.com/v1'
+    changeOrigin: true
+    pathRewrite:
+      '^/proxy/frobs/': '/'
+```
+
+```ts
+// Inside your component
+const backendUrl = config.getString('backend.baseUrl');
+fetch(`${backendUrl}/proxy/frobs/list`)
+  .then(response => response.json())
+  .then(payload => setFrobs(payload as Frob[]));
+```
+
+The proxy is powered by the `http-proxy-middleware` package, and supports all of
+its
+[configuration options](https://github.com/chimurai/http-proxy-middleware#options).
+
+Internally at Spotify, the proxy option has been the overwhelmingly most popular
+choice for plugin makers. Since we have DNS based service discovery in place and
+a microservices framework that made it trivial to expose plain HTTP, it has been
+a matter of just adding a few lines of Backstage config to get the benefit of
+being easily and robustly reachable from users' web browsers as well.
+
+This may be used instead of direct requests, when:
+
+- You need to perform HTTPS termination and/or CORS handling, because the API
+  itself is not supplying those.
+- You need to inject a simple static secret into the requests, e.g. an
```

Comment:
How do I do that? Link to docs that describe it perhaps.

Reply:
Adding headers is described in the http-proxy-middleware docs linked above